use_cases/init-process-for-fluent-bit/README.md

```diff
@@ -0,0 +1,227 @@
+
+```
+"environment": [
+                {
+                    "name": "aws_fluent_bit_init_s3_1",
+                    "value": "arn:aws:s3:::yourBucket/your-filter.conf"
+                },
+                {
+                    "name": "aws_fluent_bit_init_s3_2",
+                    "value": "arn:aws:s3:::yourBucket/your-parser.conf"
+                },
+                {
+                    "name": "aws_fluent_bit_init_s3_3",
+                    "value": "arn:aws:s3:::yourBucket/your-s3-output.conf"
+                }
+            ]
+```
+
+**your-filter.conf** (new config you created and init process downloaded from S3, path:`/init/fluent-bit-init-s3-files/your-filter.conf`)
+
+```
+[FILTER]
+    Name          parser
+    Match         xxxx
+    Key_Name      data
+    Parser        example
+```
+
+**your-parser.conf** (new config you created and init process downloaded from S3, path:`/init/fluent-bit-init-s3-files/your-parser.conf`)
+
+```
+[PARSER]
+    Name          example
+    Format        regex
+    Regex         ^(?<INT>[^ ]+) (?<FLOAT>[^ ]+) (?<BOOL>[^ ]+) (?<STRING>.+)$
+```
+
+**your-s3-output.conf** (new config you created and init process downloaded from S3, path:`/init/fluent-bit-init-s3-files/your-s3-output.conf`)
+
+```
+[OUTPUT]
+    Name                         s3
+    Match                        xxxx
+    bucket                       your-result
+    region                       ${AWS_REGION}
+    total_file_size              1M
+    upload_timeout               1m
+    use_put_object               On
+```
+
+**fluent-bit.conf** (original main config file FireLens generated, path: `/fluent-bit/etc/fluent-bit.conf`)
+
+```
+[INPUT]
+    Name       forward
+    unix_path  /var/run/fluent.sock
+
+.
+.
+.
+
+
+[OUTPUT]
+    Name 								cloudwatch
+    Match 							app-firelens*
+    auto_create_group 	true
+    log_group_name 			/aws/ecs/containerinsights/$(ecs_cluster)/application
+    log_stream_name 		$(ecs_task_id)
+    region 							us-west-2
+    retry_limit 				2
+```
+
+
+
+**You don't need to set @INCLUDE or specify the path of the parser file under the [SERVICE] section, don't need to modify the Fluent Bit command, and don't need to tell us if there is a parser config.**
+
+**The init process will download config files you specified in Task Definition from S3, and check if each config file is a parser config. If it's not a parser config, the init process will use @INCLUDE keyword to add this config to the main config file. If it's a parser config, the init process will change the original Fluent Bit command, add -R to specify that parser config.** 
+
+
+
+**After init process has processed:**
+
+**fluent-bit-init.conf** (new main config file generated by init process, will used to invoke Fluent Bit, path:`/init/fluent-bit-init.conf`)
+
+```
+@INCLUDE /fluent-bit/etc/fluent-bit.conf
+@INCLUDE /init/fluent-bit-init-s3-files/your-filter.conf
+@INCLUDE /init/fluent-bit-init-s3-files/your-parser.conf
```

Comment:
wait I think you don't include the parser... I think that goes into the command...

Reply:
Yeah, already modified.